api/services/execution/execution.proto

```diff
@@ -3,116 +3,54 @@ syntax = "proto3";
 package containerd.v1.services;
 
 import "google/protobuf/empty.proto";
+import "google/protobuf/any.proto";
 import "gogoproto/gogo.proto";
+import "github.com/docker/containerd/api/types/mount/mount.proto";
 import "github.com/docker/containerd/api/types/container/container.proto";
-import "github.com/docker/containerd/api/types/process/process.proto";
 
-service ExecutionService {
-	rpc CreateContainer(CreateContainerRequest) returns (CreateContainerResponse);
-	rpc StartContainer(StartContainerRequest) returns (google.protobuf.Empty);
-	rpc UpdateContainer(UpdateContainerRequest) returns (google.protobuf.Empty);
-	rpc PauseContainer(PauseContainerRequest) returns (google.protobuf.Empty);
-	rpc ResumeContainer(ResumeContainerRequest) returns (google.protobuf.Empty);
-	rpc DeleteContainer(DeleteContainerRequest) returns (google.protobuf.Empty);
-	rpc GetContainer(GetContainerRequest) returns (GetContainerResponse);
-	rpc ListContainers(ListContainersRequest) returns (ListContainersResponse);
-
-	rpc StartProcess(StartProcessRequest) returns (StartProcessResponse);
-	rpc GetProcess(GetProcessRequest) returns (GetProcessResponse);
-	rpc SignalProcess(SignalProcessRequest) returns (google.protobuf.Empty);
-	rpc DeleteProcess(DeleteProcessRequest) returns (google.protobuf.Empty);
-	rpc ListProcesses(ListProcessesRequest) returns (ListProcessesResponse);
-}
-
-message StartContainerRequest {
-	string id = 1 [(gogoproto.customname) = "ID"];
+service ContainerService {
+	rpc Create(CreateRequest) returns (CreateResponse);
+	rpc Start(StartRequest) returns (google.protobuf.Empty);
+	rpc Delete(DeleteRequest) returns (google.protobuf.Empty);
+	rpc List(ListRequest) returns (ListResponse);
+	rpc Events(EventsRequest) returns (stream containerd.v1.types.Event);
 }
```

Comment:
Right now you can kill via the pid.  When I port over the windows code i'll probably have to add APIs for signaling via containerd